crates/router/src/core/payment_methods/cards.rs

```diff
@@ -5516,6 +5517,12 @@ pub async fn get_bank_account_connector_details(
 
                 Ok(Some(token_data))
             }
+            PaymentMethodsData::WalletDetails(_) => {
+                Err(errors::ApiErrorResponse::UnprocessableEntity {
+                    message: "Wallet is not a valid entity".to_string(),
+                }
+                .into())
+            }
```

Comment:
Can we move this variant above the Bank details one? 
Let's have all the invalid ones together.